content/tutorials/permissionless-paymaster/10.index.md

```diff
@@ -0,0 +1,306 @@
+
+2. Dapp calls backend server or Zyfi API with relevant data to get the signer's signature.
+    - It is recommended that signer's signing part is done on a secure backend server of the Dapp.
+
+3. The signer's key signs this paymaster data and returns the signature and signer address to the Dapp's frontend.
+
+4. Paymaster address and required data with signature is added to the transaction blob in the frontend.
+
+5. User gets transaction signature request pop-up on their wallet. User only signs the transaction and transaction is sent on-chain.
+
+6. The paymaster validates the signature, identifies the manager related to the signer,
+deducts gas fees from the manager's balance, and pays for the user's transaction
+
+![flow](/images/permissionless-paymaster/flowDiagram.jpg)
+
+For this tutorial, we will use paymaster deployed on ZKsync sepolia testnet : [0xc1B0E2edC4cCaB51A764D7Dd8121CBf58C4D9E40](https://sepolia.explorer.zksync.io/address/0xc1B0E2edC4cCaB51A764D7Dd8121CBf58C4D9E40#transactions)
+
+## 1. Create a signer
+The paymaster will verify signature based on this signer address.
+The private key of this signer address should be stored securely by the Dapp.
+
+- Here is an easy way to create one:
+
+```javascript
+    import { Wallet } from "zksync-ethers";
+    const signer = Wallet.createRandom();
+```
+
+## 2. Deposit gas funds and add the signer address
+Navigate to ZKsync Sepolia Testnet Explorer: [0xc1B0E2edC4cCaB51A764D7Dd8121CBf58C4D9E40](https://sepolia.explorer.zksync.io/address/0xc1B0E2edC4cCaB51A764D7Dd8121CBf58C4D9E40#contract).
+Call `depositAndAddSigner()` function with 0.01 ether and the signer address.
+![deposit and add a signer](/images/permissionless-paymaster/depositAndAddSigner.png)
+
+::callout{icon="i-heroicons-exclamation-circle"}
+The depositor address is considered a "manager". A manager can deposit/withdraw gas funds and add/remove signers at any given time.
+
+*A manager can be a signer address too. (not recommended)*
+::
+
+## 3. Create function to sign EIP-712 typed paymaster data
+This paymaster verifies the signature signed by the signer address on the below data.
+On successful validation, it allows sponsorship for the user using manager's deposited gas funds.
+
+```solidity
+        bytes32 messageHash = keccak256(
+            abi.encode(
+                SIGNATURE_TYPEHASH,
+                _from,
+                _to,
+                _expirationTime,
+                _maxNonce,
+                _maxFeePerGas,
+                _gasLimit
+            )
+        );
+```
+
+- Create a `getSignature()` function in your backend(recommended)
+
+```javascript
+// Example code
+import {BigNumber, Contract, Wallet} from "zksync-ethers";
+export async function getSignature(
+  from: string, to: string, expirationTime: BigNumber, maxNonce: BigNumber, maxFeePerGas: BigNumber, gasLimit: BigNumber, paymaster: Contract
+){
+  const signer = new Wallet(process.env.SIGNER_PRIVATE_KEY, provider);
```

Comment:
Can you clarify that the reader will need to have the provider and .env file set up for this? 

Could also add something like 

```ts
const rpcUrl = process.env.ZKSYNC_RPC_URL ?? 'https://sepolia.era.zksync.dev';
  const provider = new Provider(rpcUrl);
```

Reply:
Updated with provider and added more clarity on step-3